packages/synapse-sdk/src/telemetry/service.ts

```diff
@@ -0,0 +1,218 @@
+    this.sentry.setTag('runtime', this.context.runtime) // only two values, useful for searching
+    this.sentry.setTag('network', this.context.network as 'mainnet' | 'calibration') // only two values, useful for searching
+  }
+
+  protected sanitizeError(event: any): any {
+    this.addToBuffer(event)
+
+    return event
+  }
+
+  /**
+   * Get debug dump for support tickets
+   *
+   * Returns enough information for devs to dive into the data on filoz.sentry.io
+   *
+   * @example
+   * ```typescript
+   * const dump = synapse.telemetry.debugDump()
+   * console.log(JSON.stringify(dump, null, 2))
+   * ```
+   */
+  debugDump(limit = 50): DebugDump {
+    return {
+      lastEventId: this.sentry.lastEventId(),
+      events: this.eventBuffer.slice(-limit),
+    }
+  }
+
+  /**
+   * Check if telemetry is enabled
+   */
+  isEnabled(): boolean {
+    return this.enabled
+  }
+
+  /**
+   * Enable telemetry explicitly (even if disabled by environment)
+   * Useful for testing or when you need to force telemetry on
+   */
+  enable(): void {
+    if (!this.enabled) {
+      this.enabled = true
+      this.initSentry()
+    }
+  }
+
+  /**
+   * Disable telemetry explicitly (even if enabled by environment)
+   * Useful for testing or when you need to force telemetry off
+   */
+  disable(ms?: number): void {
+    if (this.enabled) {
+      this.enabled = false
+      void this.sentry.close(ms)
+    }
+  }
+
+  /**
+   * Flush pending telemetry events
+   *
+   * Call this before process exit to ensure all events are sent.
+   * Returns a promise that resolves when flushing is complete.
+   *
+   * @param timeout - Maximum time to wait in milliseconds (default: 2000ms)
+   * @returns Promise that resolves to true if all events were flushed
+   *
+   * @example
+   * ```typescript
+   * // Before exiting
+   * await synapse.telemetry.flush()
+   * process.exit(0)
+   * ```
+   */
+  async flush(timeout = 2000): Promise<boolean> {
+    if (!this.enabled) {
+      return true
+    }
+
+    // Delegate to adapter's flush method if available
+    return this.sentry.flush(timeout)
+  }
+
+  /**
+   * Close the telemetry service and shut down the adapter
+   *
+   * This flushes pending events and closes the telemetry connection.
+   * Call this when your application is shutting down to ensure proper cleanup.
+   *
+   * @param timeout - Maximum time to wait in milliseconds (default: 2000ms)
+   * @returns Promise that resolves to true if shutdown was successful
+   *
+   * @example
+   * ```typescript
+   * // Before exiting
+   * await synapse.telemetry.close()
+   * process.exit(0)
+   * ```
+   */
+  async close(timeout = 2000): Promise<boolean> {
+    if (!this.enabled) {
+      return true
+    }
+
+    return this.sentry.close(timeout)
+  }
+
+  /**
+   * Add event to circular buffer
+   * @internal
+   */
+  private addToBuffer(event: any): void {
```

Comment:
```suggestion
  private addToEventBuffer(event: any): void {
```

to be a little more self describing?